pdf/differential_test/main.py

```diff
@@ -6,208 +6,178 @@
 import io
 import numpy as np
 import os
-import scipy.ndimage
+import scipy
 import shutil
 import subprocess
 import sys
 import tempfile
 
-DPI = 100  # Too low, and aliasing will make some colors unreachable.
 RMSE_WINDOW_SIZE = 50
-RMSE_FOR_SCHIZO = 30.  # Out of 255
-DIFF_FOR_SCHIZO = -12.  # Negative means disabled
-
-tools = [
-        # File prefix must always be f'{name}-{page}.ext'
-        # Can optionally take 'env'
-        # Lambda function takes output directory
-        # '<inputFile>' replaced with input file name.
-        {
-            'name': 'mutool',
-            'exec': ['mutool', 'draw', '-r', str(DPI), '-o',
-                lambda dname: os.path.join(dname, 'mutool-%d.png'),
-                '<inputFile>'],
-        },
-        {
-            'name': 'pdftoppm',
-            'exec': ['pdftoppm', '-png', '-r', str(DPI),
-                # PDFs can specify a "crop box" apparently. These files render
-                # with a draft border or other issue if this is not specified,
-                # preventing the render from matching e.g. mutool.
-                '-cropbox',
-                '<inputFile>',
-                lambda dname: os.path.join(dname, 'pdftoppm')],
-        },
-        {
-            'name': 'xpdf-pdftoppm',
-            'exec': ['/opt/xpdf/bin64/pdftoppm', '-r', str(DPI),
-                #'-cropbox',
-                '<inputFile>',
-                lambda dname: os.path.join(dname, 'xpdf-pdftoppm')],
-        },
-]
+RMSE_FOR_DIFFERENTIAL = 30.  # Out of 255
+DIFF_FOR_DIFFERENTIAL = -12.  # Negative means disabled
+
 
 def main():
-    fname = sys.argv[1]
+    artifacts_root_dir = sys.argv[1]
     html_out = len(sys.argv) > 2 and sys.argv[2] == '--html'
 
-    tool_names = [tool['name'] for tool in tools]
-
     img_attrs = 'width="400"'
-    with TempDir() as dname:
-        for t in tools:
-            ex = t['exec']
-            ex = [e if e != '<inputFile>' else fname for e in ex]
-            ex = [e if not callable(e) else e(dname) for e in ex]
-            env = os.environ.copy()
-            env.update(t.get('env', {}))
-            subprocess.check_call(ex, env=env, stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE)
+
+    tool_names = []
+    pages_by_tool = {}
+    for t in os.listdir(artifacts_root_dir):
```

Comment:
`sorted(os.listdir(...))` would be slightly more compact than the separate `tool_names.sort` call